### PR TITLE
Update default search for 5 regions

### PR DESCRIPTION
**Pull request by bsclifton**
Fixes https://github.com/brave/brave-browser/issues/18331
Fixes https://github.com/brave/brave-browser/issues/18415
Fixes https://github.com/brave/brave-browser/issues/18452

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used GitHub [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: android/java/org/chromium/chrome/browser/toolbar/top/BraveToolbarLayoutImpl.java [hunk 1]
```diff
@@ -132,6 +133,8 @@
                    BraveRewardsObserver, BraveRewardsNativeWorker.PublisherObserver {
     public static final String PREF_HIDE_BRAVE_REWARDS_ICON = "hide_brave_rewards_icon";
     private static final String JAPAN_COUNTRY_CODE = "JP";
+    private static final List<String> mBraveSearchEngineDefaultRegions =
+            Arrays.asList("CA", "DE", "FR", "GB", "US");
```
Comment (bsclifton) [inline android/java/org/chromium/chrome/browser/toolbar/top/BraveToolbarLayoutImpl.java:137]: Besides these, I think we also need remove the `US`/`CA` restriction we added in https://github.com/brave/brave-core/pull/8966/commits/4808157d3bab53e5c0509808f20c295ee124e247 (to `android/java/org/chromium/chrome/browser/onboarding/SearchEngineOnboardingFragment.java`)

That way we can show Brave Search for all other regions (which don't have Brave Search as default)

**Review by goodov — APPROVED** [review]
`chromium_src` lgtm. in general also lgtm.

File: android/java/org/chromium/chrome/browser/toolbar/top/BraveToolbarLayoutImpl.java (hunk 1 above)
Comment (bsclifton) [reply android/java/org/chromium/chrome/browser/toolbar/top/BraveToolbarLayoutImpl.java:137]: Actually - followed up with @ecnmst - no change needed here for now 😄

**Review by emerick — APPROVED** [review]
LGTM, just had one minor nit.

**Comment by kjozwiak** [comment]
Verification **PASSED** on `Win 11 x64` using the following build:

```
Brave | 1.32.61 Chromium: 95.0.4638.40 (Official Build) nightly (64-bit)
-- | --
Revision | e3e7c76ba0284b16087cf4cf3153abfaef6470c7-refs/branch-heads/4638@{#624}
OS | Windows 11 Version 21H2 (Build 22000.194)
```

### Case https://github.com/brave/brave-browser/issues/18415 (Removed from `brave://welcome`)

* ensured that `CA`, `USA`, `GB`, `DE` & `FR` had the search section removed from `brave://welcome`

**Search section being removed from `brave://welcome`**

**`CA`** | **`USA`** | **`GB`** | **`DE`** | **`FR`**
---------|------------|----------|----------|-------
Canada | USA | GB | Germany | France

**Search section shouldn't be removed from `brave://welcome`**

**`Poland`** | **`Spain`**
--------------|-----------
Poland |Spain

**Pull request merged**